File: alumnium/tools/click_tool.py
from pydantic import Field

from alumnium.drivers import BaseDriver
from .base_tool import BaseTool


class ClickTool(BaseTool):
    """Click an element."""

    id: int = Field(description="Element identifier (ID)")

    def invoke(self, driver: BaseDriver):
        driver.click(self.id)

File: packages/typescript/src/tools/ClickTool.ts
import { BaseDriver } from "../drivers/BaseDriver.js";
import { BaseTool } from "./BaseTool.js";
import { field, FieldMetadata } from "./Field.js";
import { SelectTool } from "./SelectTool.js";

export class ClickTool extends BaseTool {
  static description = `Click an element. Avoid using this tool for combobox dropdowns; use ${SelectTool.name} instead.`;
  static fields: FieldMetadata[] = [
    field({
      name: "id",
      type: "integer",
      description: "Element identifier (ID)",
    }),
  ];

  id: number;

  constructor(args: { id: number }) {
    super();
    this.id = args.id;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.click(this.id);
  }
}
